tuxemon/technique.py

```diff
@@ -297,25 +300,38 @@ def changestats(self, user: Monster, target: Monster) -> EffectResult:
         '''
         statsmaster = [self.statspeed, self.stathp, self.statarmour, self.statmelee, self.statranged, self.statdodge]
         statslugs = ['speed', 'hp', 'armour', 'melee', 'ranged', 'dodge']
-        for stat, slug in zip(statsmaster, statslugs):
+        newstatvalue = 0
+        for stat, slugdata in zip(statsmaster, statslugs):
             if not stat:
                 continue
-            value = stat['value']
-            dividing = stat['dividing']
-            override = stat['overridetofull']
-            basestatvalue = getattr(target, slug)
-            if value > 0 and override == False:
-                if dividing == False:
-                    newstatvalue = basestatvalue + value
-                else:
-                    newstatvalue = basestatvalue // value
-                success = True
-                setattr(target, slug, newstatvalue)
-            if override == True and slug == 'hp':
-                success = True
-                target.current_hp = target.hp
+            value = stat.get('value')
+            max_deviation = stat.get('max_deviation')
+            operation = stat.get('operation')
+            override = stat.get('overridetofull')
+            if not override:
+                override = False
+            if not max_deviation:
+                max_deviation = False
+            basestatvalue = getattr(target, slugdata)
+            if max_deviation:
+                value = random.randint(value - max_deviation, value + max_deviation)
+                
+            if value > 0 and override != True:
+                ops_dict = {
+                "+": operator.add,
+                "-": operator.sub,
+                "*": operator.mul,
+                "/": operator.floordiv,
+                }
+
+                newstatvalue = ops_dict[operation](basestatvalue, value)
```

Comment:
need to try/except for divide by zero errors.  though unlikely, if the value is zero, and operation is 0, there will be a crash here

Reply:
i dont think thatll happen because the condition above it says to only happen if value >0